File: Examples/VisualBasic/Shapes/ConnectorLineAngle.vb
Imports System

Namespace Aspose.Slides.Examples.VisualBasic.Shapes
    Public Class ConnectorLineAngle
        Public Shared Sub Run()
			'ExStart:ConnectorLineAngle
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Shapes()

            Dim pres As New Presentation(dataDir & "ConnectorLineAngle.pptx")
            Dim slide As Slide = CType(pres.Slides(0), Slide)
            Dim shape As Shape
            For i As Integer = 0 To slide.Shapes.Count - 1
                Dim dir As Double = 0.0
                shape = CType(slide.Shapes(i), Shape)
                If TypeOf shape Is AutoShape Then
                    Dim ashp As AutoShape = CType(shape, AutoShape)
                    If ashp.ShapeType = ShapeType.Line Then
                        dir = getDirection(ashp.Width, ashp.Height, Convert.ToBoolean(ashp.Frame.FlipH), Convert.ToBoolean(ashp.Frame.FlipV))
                    End If
                ElseIf TypeOf shape Is Connector Then
                    Dim ashp As Connector = CType(shape, Connector)
                    dir = getDirection(ashp.Width, ashp.Height, Convert.ToBoolean(ashp.Frame.FlipH), Convert.ToBoolean(ashp.Frame.FlipV))
                End If

                Console.WriteLine(dir)
            Next i

        End Sub
        Public Shared Function getDirection(ByVal w As Single, ByVal h As Single, ByVal flipH As Boolean, ByVal flipV As Boolean) As Double
            Dim endLineX As Single = w * (If(flipH, -1, 1))
            Dim endLineY As Single = h * (If(flipV, -1, 1))
            Dim endYAxisX As Single = 0
            Dim endYAxisY As Single = h
            Dim angle As Double = (Math.Atan2(endYAxisY, endYAxisX) - Math.Atan2(endLineY, endLineX))
            If angle < 0 Then
                angle += 2 * Math.PI
            End If
            Return angle * 180.0 / Math.PI
        End Function
		
		'ExEnd:ConnectorLineAngle
    End Class

End Namespace

File: Examples/CSharp/Shapes/ConnectorLineAngle.cs
using System.IO;
using Aspose.Slides;
using System;

namespace Aspose.Slides.Examples.CSharp.Shapes
{
    public class ConnectorLineAngle
    {
        //ExStart:ConnectorLineAngle
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            Presentation pres = new Presentation(dataDir + "ConnectorLineAngle.pptx");
            Slide slide = (Slide)pres.Slides[0];
            Shape shape;
            for (int i = 0; i < slide.Shapes.Count; i++)
            {
                double dir = 0.0;
                shape = (Shape)slide.Shapes[i];
                if (shape is AutoShape)
                {
                    AutoShape ashp = (AutoShape)shape;
                    if (ashp.ShapeType == ShapeType.Line)
                    {
                        dir = getDirection(ashp.Width, ashp.Height, Convert.ToBoolean(ashp.Frame.FlipH), Convert.ToBoolean(ashp.Frame.FlipV));
                    }
                }
                else if (shape is Connector)
                {
                    Connector ashp = (Connector)shape;
                    dir = getDirection(ashp.Width, ashp.Height, Convert.ToBoolean(ashp.Frame.FlipH), Convert.ToBoolean(ashp.Frame.FlipV));
                }

                Console.WriteLine(dir);
            }

        }
        public static double getDirection(float w, float h, bool flipH, bool flipV)
        {
            float endLineX = w * (flipH ? -1 : 1);
            float endLineY = h * (flipV ? -1 : 1);
            float endYAxisX = 0;
            float endYAxisY = h;
            double angle = (Math.Atan2(endYAxisY, endYAxisX) - Math.Atan2(endLineY, endLineX));
            if (angle < 0) angle += 2 * Math.PI;
            return angle * 180.0 / Math.PI;
        }
        //ExEnd:ConnectorLineAngle
    }

}